common/djangoapps/enrollment/tests/test_views.py

```diff
@@ -1240,3 +1241,162 @@ def _cross_domain_post(self, csrf_cookie):
+
+    def _make_request(self, query_params=None):
+        return self.client.get(self.url, query_params)
+
+    def _assert_list_of_enrollments(self, query_params=None, expected_status=status.HTTP_200_OK, error_fields=None):
+        response = self._make_request(query_params)
+        self.assertEqual(response.status_code, expected_status)
+        content = json.loads(response.content)
+        if error_fields is not None:
+            self.assertIn('field_errors', content)
+            for error_field in error_fields:
+                self.assertIn(error_field, content['field_errors'])
+        return content
+
+    def test_user_not_authenticated(self):
+        self.client.logout()
+        response = self.client.get(self.url, {'course_id': self.course.id})
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_user_not_authorized(self):
+        self.client.login(username=self.student1.username, password='edx')
+        response = self.client.get(self.url, {'course_id': self.course.id})
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    @ddt.data(
+        ({'course_id': '1'}, ['course_id', ]),
+        ({'course_id': '1', 'username': 'staff'}, ['course_id', ]),
+        ({'username': '1*2'}, ['username', ]),
+        ({'username': '1*2', 'course_id': 'org.0/course_0/Run_0'}, ['username', ]),
+        ({'username': '1*2', 'course_id': '1'}, ['username', 'course_id'])
+    )
+    @ddt.unpack
+    def test_query_string_parameters_invalid_errors(self, query_params, error_fields):
+        self._login_as_staff()
+        self._assert_list_of_enrollments(query_params, status.HTTP_400_BAD_REQUEST, error_fields)
+
+    @ddt.data(
+        # Non-existent user
+        ({'username': 'nobody'}, ),
+        ({'username': 'nobody', 'course_id': 'e/d/X'}, ),
+
+        # Non-existent course
+        ({'course_id': 'a/b/c'}, ),
+        ({'course_id': 'a/b/c', 'username': 'student1'}, ),
+
+        # Non-existent course and user
+        ({'course_id': 'a/b/c', 'username': 'dummy'}, )
+    )
+    @ddt.unpack
+    def test_non_existent_course_user(self, query_params):
+        self._login_as_staff()
+        content = self._assert_list_of_enrollments(query_params, status.HTTP_200_OK)
+        self.assertEqual(len(content['results']), 0)
+
+    @ddt.file_data('fixtures/course-enrollments-api-list-valid-data.json')
+    @ddt.unpack
+    def test_response_valid_queries(self, args):
+        query_params = args[0]
+        expected_results = args[1]
+
+        self._login_as_staff()
+        content = self._assert_list_of_enrollments(query_params, status.HTTP_200_OK)
+        results = content['results']
+        self.assertEqual(len(results), len(expected_results))
+
+        expected_fields = ('course_id', 'user', 'mode', 'created', 'is_active')
+        for item in content['results']:
+            self.assertTrue(any(field in item for field in expected_fields))
```

Comment:
nit: you can even get rid of these 3 lines, because the `assertItemsEqual` check below does the same thing.

Reply:
@pomegranited even the length check above can be removed. Doing it now

Reply:
Done.